code/modules/admin/verbs/randomverbs.dm

```diff
@@ -349,111 +349,117 @@ Traitors and the like can also be revived with the previous role mostly intact.
+		new_character = new(srcturf)
+
+	else if(location == "Arrivals") //Spawn them at a latejoin spawnpoint
+		new_character = new(pick(latejoin))
+
+	else //I have no idea how you're here
+		return
+
+	if(!new_character)
+		src << "Something went wrong and spawning failed."
+		return
+
+	//Write the appearance and whatnot out to the character
+	picked_client.prefs.copy_to(new_character)
+	if(inhabit)
+		new_character.key = player_key
+
+	//Were they any particular special role? If so, copy.
 	var/datum/antagonist/antag_data = get_antag_data(new_character.mind.special_role)
 	if(antag_data)
 		antag_data.add_antagonist(new_character.mind)
 		antag_data.place_mob(new_character)
-	else
-		job_master.EquipRank(new_character, new_character.mind.assigned_role, 1)
 
-	//Announces the character on all the systems, based on the record.
-	if(!issilicon(new_character))//If they are not a cyborg/AI.
-		if(!record_found && !player_is_antag(new_character.mind, only_offstation_roles = 1)) //If there are no records for them. If they have a record, this info is already in there. MODE people are not announced anyway.
-			//Power to the user!
-			if(alert(new_character,"Warning: No data core entry detected. Would you like to announce the arrival of this character by adding them to various databases, such as medical records?",,"No","Yes")=="Yes")
-				data_core.manifest_inject(new_character)
+	//Referenced to later to apply previous settings (role, antag) if any
+	var/datum/data/record/record_found
+	var/formerjob = "Assistant"
+	if(new_character.mind)
+		/*Try and locate a record for the person being respawned through data_core.
+		This isn't an exact science but it does the trick more often than not.*/
+		record_found = find_general_record("name",new_character.real_name)
+		if(record_found)
+			formerjob = record_found.fields["real_rank"]
+			new_character.mind.assigned_role = formerjob
+
+	//If they had a job before, re-equip them for their job.
+	if(equipment)
+		job_master.EquipRank(new_character, formerjob, 1)
 
-			if(alert(new_character,"Would you like an active AI to announce this character?",,"No","Yes")=="Yes")
-				call(/proc/AnnounceArrival)(new_character, new_character.mind.assigned_role)
+	//A redraw for good measure
+	new_character.update_icons()
 
-	message_admins("\blue [admin] has respawned [player_key] as [new_character.real_name].", 1)
+	//If we're announcing their arrival
+	if(announce)
+		AnnounceArrival(new_character, new_character.mind.assigned_role)
 
-	new_character << "You have been fully respawned. Enjoy the game."
+	//Announces the character on all the systems, based on the record.
+	if(!record_found)
+		if(alert(new_character,"Warning: No data core entry detected. Would you like add them to various databases, such as medical records?","Records","Yes","No")=="Yes")
+			data_core.manifest_inject(new_character)
+
+	message_admins("\blue [admin] has spawned [player_key]'s character [new_character.real_name].", 1)

```

Comment:
S'how the original was, was just copypaste, but what is a better proc to use? I don't really recall having to use span classes in other places with admin messages, I thought it auto-colored them. Is there some other, better admin-message proc?